--- a/mt7915/mt7915.h
+++ b/mt7915/mt7915.h
@@ -838,6 +838,27 @@ int mt7915_mmio_wed_init(struct mt7915_d
 			 bool pci, int *irq);
 
 #ifdef CONFIG_MTK_VENDOR
+struct mt7915_muru;
+struct mt7915_tm_rf_test_result;
+struct mt7915_tm_rf_test_data;
+
+void mt7915_vow_init(struct mt7915_dev *dev);
+int mt7915_get_psd_country(char *country);
+int mt7915_mcu_set_muru_cfg(struct mt7915_phy *phy, struct mt7915_muru *muru);
+int mt7915_mcu_set_bsrp_ctrl(struct mt7915_phy *phy, u16 interval, u16 duration,
+			     u32 bitmap, u8 band_idx, u8 enable);
+int mt7915_mcu_set_txbf_sound_info(struct mt7915_phy *phy, u8 action, u8 bw,
+				   u8 band_idx, u8 enable);
+void mt7915_sta_scs_para(void *data, struct ieee80211_sta *sta);
+int mt7915_mcu_set_scs_stats(struct mt7915_phy *phy);
+u32 vow_chip_show_sta_acq_info(struct mt7915_dev *dev, u32 *ple_stat,
+			       u32 *sta_pause, u32 *dis_sta_map,
+			       u32 dumptxd);
+int mt7915_vow_pleinfo_read(struct mt7915_dev *dev);
+void mt7915_tm_re_cal_event(struct mt7915_dev *dev,
+			    struct mt7915_tm_rf_test_result *result,
+			    struct mt7915_tm_rf_test_data *data);
+
 void mt7915_set_wireless_amsdu(struct ieee80211_hw *hw, u8 en);
 void mt7915_capi_sta_rc_work(void *data, struct ieee80211_sta *sta);
 void mt7915_set_wireless_vif(void *data, u8 *mac, struct ieee80211_vif *vif);
